### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by hey-waffles**

### CODE REVIEW

File: app/src/pages/admin/game-systems/new.tsx [hunk 1]
```diff
@@ -63,45 +51,33 @@ export function NewGameSystemForm(props: GameSystemFormProps) {
...
-    id,
-    name,
-  }
-}
-`;
-
 NewGameSystem.getInitialProps = async () => {
   // const { themes } = await client.query({query: fetchThemesGQL});
-  return { themes: [{"name": "Default", "id": "default"}]};
+  // return { themes: [{"name": "Default", "id": "default"}]};
```
Comment (hey-waffles) [inline app/src/pages/admin/game-systems/new.tsx:80]: delete

File: app/src/pages/admin/test-pages/misc.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import React, { ReactNode } from "react";
 import Page from "../../../components/design/Page";
 import { Button } from "react-bootstrap";
 import Tooltip from "../../../components/design/Tooltip";
 
 /**
  * A page devoted to miscellaneous component testing
  */
-export default function Miscellaneous() {
+export default function Miscellaneous(): ReactNode {
```
Comment (hey-waffles) [inline app/src/pages/admin/test-pages/misc.tsx:9]: JSX.Element

File: app/src/pages/admin/test-pages/tables.tsx [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-  return <><Link href={"/game-systems/" + props.key + "/edit"} passHref><a>Edit</a></Link></>
+function TestTableActions({ data }: TableComponentProps) {
+  return <><Link href={"/game-systems/" + data.key + "/edit"} passHref><a>Edit</a></Link></>
 }
 
 /**
  * Creates a test table for checking that new changes and functionality do not break and work as intended.
  */
-export default function TableTesting() {
+export default function TableTesting(): ReactNode {
```
Comment (hey-waffles) [inline app/src/pages/admin/test-pages/tables.tsx:19]: JSX.Element

File: app/src/pages/index.tsx [hunk 4]
```diff
@@ -13,7 +12,7 @@ import { signOut, useSession } from "next-auth/client";
  *
  * @param props ...
  */
-function Index(props: any) {
+function Index(): ReactNode {
```
Comment (hey-waffles) [inline app/src/pages/index.tsx:15]: JSX.ELement

File: app/src/server/resolvers/CoreResolver.ts [hunk 5]
```diff
@@ -156,13 +145,14 @@ export class CoreResolver {
...
-        superDocumentAliases[superDocument],
-        superDocumentAliasModels[superDocument]
+        (superDocumentAliases as Record<string, string>)[superDocument],
+        (superDocumentAliasModels as Record<string, unknown>)[superDocument]
       );
 
       if (!superDocumentResult) { return null; }
-
-      filters[`${superDocument}_eq`] = superDocumentResult._id;
+
```
Comment (hey-waffles) [inline app/src/server/resolvers/CoreResolver.ts:153]: Lots of newlines?

File: app/src/utilities/tools.ts [hunk 6]
```diff
@@ -136,34 +25,17 @@ export function min(arg: number, defaultValue: number): number {
  * @param obj The object to keep fields from
  * @param fields The fields to keep
  */
-export function objectKeepFields(obj: any, fields: string[]) {
-  const newObject: any = {};
+export function objectKeepFields(obj: Record<string, unknown>, fields: string[]): Record<string, unknown> {
```
Comment (hey-waffles) [inline app/src/utilities/tools.ts:28]: Check that this is actually used somewhere

File: model/src/filters/index.ts [hunk 7]
```diff
@@ -13,3 +13,52 @@ export * from "./Module";
 export * from "./Organization";
 export * from "./Rule";
 export * from "./User";
+
```
Comment (hey-waffles) [inline model/src/filters/index.ts:16]: Comment these

File: model/src/documents/index.ts [hunk 8]
```diff
@@ -13,3 +13,54 @@ export * from "./Module";
...
+import { Entity, EntityModel } from "./Entity";
+import { EntityLayout, EntityLayoutModel } from "./EntityLayout";
+import { EntityType, EntityTypeModel } from "./EntityType";
+import { GameSystem, GameSystemModel } from "./GameSystem";
+import { Module, ModuleModel } from "./Module";
+import { Organization, OrganizationModel } from "./Organization";
+import { Rule, RuleModel } from "./Rule";
+import { User, UserModel } from "./User";
+
+export type GenericDocumentType = (
```
Comment (hey-waffles) [inline model/src/documents/index.ts:33]: Comment

File: model/src/inputs/index.ts [hunk 9]
```diff
@@ -13,3 +13,54 @@ export * from "./Module";
...
+import { CreateEntityTypeInput, UpdateEntityTypeInput } from "./EntityType";
+import { CreateGameSystemInput, UpdateGameSystemInput } from "./GameSystem";
+import { CreateModuleInput, UpdateModuleInput } from "./Module";
+import { CreateOrganizationInput, UpdateOrganizationInput } from "./Organization";
+import { CreateRuleInput, UpdateRuleInput } from "./Rule";
+import { CreateUserInput, UpdateUserInput } from "./User";
+
+export type GenericInputType = GenericCreateInputType | GenericUpdateInputType;
+
+export type GenericCreateInputType = (
```
Comment (hey-waffles) [inline model/src/inputs/index.ts:34]: Comment

File: app/package.json [hunk 10]
```diff
@@ -8,7 +8,8 @@
     "build-all": "lerna run build --scope=@reroll/app --include-dependencies",
     "postbuild": "next-on-netlify",
     "start": "next start",
-    "test": "jest"
+    "test": "jest",
+    "lint": "eslint . --ext .js,.jsx,.ts,.tsx"
```
Comment (hey-waffles) [inline app/package.json:12]: Delete this. Linting run from root

File: model/package.json [hunk 11]
```diff
@@ -8,7 +8,8 @@
   "private": true,
   "scripts": {
     "build": "tsc",
-    "dev": "nodemon -V -w src -e ts,tsx -x \"yarn build\""
+    "dev": "nodemon -V -w src -e ts,tsx -x \"yarn build\"",
+    "lint": "eslint . --ext .js,.jsx,.ts,.tsx"
```
Comment (hey-waffles) [inline model/package.json:12]: Delete this. Linting run from root

File: app/src/utilities/tools.ts (hunk 6 above)
Comment (hey-waffles) [reply app/src/utilities/tools.ts:28]: We don't. Removing it

**Review by lucyawrey — APPROVED** [review]
Looking good

**Pull request merged**